ray-operator/controllers/ray/raycluster_controller_fake_test.go

```diff
@@ -2380,3 +2380,65 @@ func TestSumGPUs(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteAllPods(t *testing.T) {
+	newScheme := runtime.NewScheme()
+	_ = corev1.AddToScheme(newScheme)
+	ns := "tmp-ns"
+	ts := metav1.Now()
+	filter := map[string]string{"app": "tmp"}
+
+	p1 := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "alive",
+			Namespace: ns,
+			Labels:    filter,
+		},
+	}
+	p2 := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "deleted",
+			Namespace:         ns,
+			Labels:            filter,
+			DeletionTimestamp: &ts,
+			Finalizers:        []string{"tmp"},
+		},
+	}
+	p3 := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "other",
+			Namespace: ns,
+			Labels:    map[string]string{"app": "other"},
+		},
+	}
+
+	fakeClient := clientFake.NewClientBuilder().
+		WithScheme(newScheme).
+		WithRuntimeObjects(p1, p2, p3).
+		Build()
+
+	testRayClusterReconciler := &RayClusterReconciler{
+		Client:   fakeClient,
+		Recorder: &record.FakeRecorder{},
+		Scheme:   newScheme,
+		Log:      ctrl.Log.WithName("controllers"),
+	}
+	// first deleteAllPods. it should delete the "alive" pod.
+	active, pods, err := testRayClusterReconciler.deleteAllPods(context.Background(), ns, filter)
+	assert.Nil(t, err)
+	assert.Equal(t, 1, active)
+	assert.Equal(t, 2, len(pods.Items))
+	assert.Equal(t, []string{"alive", "deleted"}, []string{pods.Items[0].Name, pods.Items[1].Name})
+	// second deleteAllPods.  it should delete no pod (active == 0).
+	active, pods, err = testRayClusterReconciler.deleteAllPods(context.Background(), ns, filter)
```

Comment:
Sure, this will make the code shorter and be more easy to read.

Typically, a `context.Background()` will be inlined to an empty struct, therefore there will be no performance impact even if you don't reuse it.